pkg/apiserver/resthandler.go

```diff
@@ -269,8 +269,14 @@ func CreateResource(r rest.Creater, scope RequestScope, typer runtime.ObjectType
 			errorJSON(err, scope.Codec, w)
 			return
 		}
+		// Get name, ok if empty
+		_, name, _ := scope.Namer.Name(req)
+
 		ctx := scope.ContextFunc(req)
 		ctx = api.WithNamespace(ctx, namespace)
+		if len(name) > 0 {
+			ctx = api.WithValue(ctx, "name", name)
```

Comment:
Post/Create is kind of unique because it doesn't take a name parameter. Not sure what you mean by adding it to the other objects. It would be like GetterWithOptions... if your storage object happens to implement it, we use it, if not, then great.